File: app/src/main/java/com/past/music/adapter/SongListAdapter.java
package com.past.music.adapter;

import android.content.Context;
import android.os.Handler;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.past.music.entity.MusicEntity;
import com.past.music.pastmusic.R;
import com.past.music.service.MusicPlayer;
import com.past.music.utils.HandlerUtil;

import java.util.HashMap;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * =======================================================
 * 作者：GaoJin
 * 日期：2017/4/23 19:51
 * 描述：热门歌单适配器
 * 备注：
 * =======================================================
 */
public class SongListAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private Context mContext;
    private List<MusicEntity> mList;
    private Handler mHandler;

    public SongListAdapter(Context context) {
        this.mContext = context;
        mHandler = HandlerUtil.getInstance(context);
    }

    public void updateList(List<MusicEntity> list) {
        this.mList = list;
        notifyDataSetChanged();
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new RecommendListHolder(LayoutInflater.from(mContext).inflate(R.layout.hot_list_item, parent, false));
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ((RecommendListHolder) holder).onBind(mList.get(position));
    }

    @Override
    public int getItemCount() {
        return mList == null ? 0 : mList.size();
    }

    class RecommendListHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        @BindView(R.id.hot_img_singer)
        SimpleDraweeView mImgSong;

        @BindView(R.id.hot_singer_list_title)
        TextView mTitle;

        @BindView(R.id.hot_singer_list_info)
        TextView mInfo;

        @BindView(R.id.rl_hot_singer_item)
        RelativeLayout relativeLayout;


        public RecommendListHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            itemView.setOnClickListener(this);
        }

        public void onBind(final MusicEntity musicEntity) {
            mTitle.setText(musicEntity.getMusicName());
            mTitle.setTextColor(0xffffffff);
            mInfo.setText(musicEntity.getArtist() + " • " + "专辑");
            mImgSong.setImageURI(musicEntity.getAlbumData());
        }

        @Override
        public void onClick(View view) {
            mHandler.postDelayed(new Runnable() {
                @Override
                public void run() {
                    HashMap<Long, MusicEntity> infos = new HashMap<Long, MusicEntity>();
//                    int len = 10;
                    int len = mList.size();
                    long[] list = new long[len];
                    for (int i = 0; i < len; i++) {
                        MusicEntity info = mList.get(i);
                        list[i] = info.songId;
                        infos.put(list[i], info);
                    }
                    if (getAdapterPosition() > 0)
                        MusicPlayer.playAll(infos, list, getAdapterPosition(), false);
                }
            }, 70);
        }
    }
}

File: app/src/main/java/com/past/music/adapter/SongListAdapter.kt
package com.past.music.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Color
import android.os.Handler
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import com.facebook.drawee.view.SimpleDraweeView
import com.past.music.dialog.NetSongDialog
import com.past.music.entity.MusicEntity
import com.past.music.pastmusic.R
import com.past.music.service.MusicPlayer
import com.past.music.utils.HandlerUtil
import com.past.music.utils.ImageUtils
import com.past.music.utils.MConstants
import java.util.*

/**
 * Created by gaojin on 2017/12/14.
 */
class SongListAdapter(context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var mContext: Context = context
    private var mList: List<MusicEntity>? = null
    private var mHandler: Handler? = null
    private var isLight = true
    private var netSongDialog: NetSongDialog? = null

    constructor(context: Context, isLight: Boolean) : this(context) {
        mHandler = HandlerUtil.getInstance(context)
        this.isLight = isLight
    }

    fun updateList(list: List<MusicEntity>) {
        this.mList = list
        notifyDataSetChanged()
    }

    init {
        mHandler = HandlerUtil.getInstance(context)
    }

    override fun getItemCount(): Int {
        return if (mList == null) 0 else mList!!.size
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder?, position: Int) {
        (holder as RecommendListHolder).onBind(mList!![position])
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): RecyclerView.ViewHolder {
        return RecommendListHolder(LayoutInflater.from(mContext).inflate(R.layout.hot_list_item, parent, false))
    }

    internal inner class RecommendListHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        var mImgSong: SimpleDraweeView = itemView.findViewById(R.id.hot_img_singer)
        var mTitle: TextView = itemView.findViewById(R.id.hot_singer_list_title)
        var mInfo: TextView = itemView.findViewById(R.id.hot_singer_list_info)
        var relativeLayout: RelativeLayout = itemView.findViewById(R.id.rl_hot_singer_item)
        var mLine: View = itemView.findViewById(R.id.line)

        fun more() {
            netSongDialog = NetSongDialog(mContext, mList!![adapterPosition], MConstants.MUSICOVERFLOW)
            netSongDialog!!.show()
        }

        init {
            itemView.setOnClickListener(this)
            itemView.findViewById<View>(R.id.three_more).setOnClickListener(this)
        }

        fun onBind(musicEntity: MusicEntity) {
            if (isLight) {
                mTitle.setTextColor(Color.WHITE)
                mLine.visibility = View.GONE
            } else {
                mTitle.setTextColor(Color.BLACK)
                relativeLayout.setBackgroundColor(Color.WHITE)
                mLine.visibility = View.VISIBLE
            }
            mTitle.text = musicEntity.getMusicName()
            mInfo.text = musicEntity.getArtist()
            ImageUtils.setImageSource(mContext, mImgSong, musicEntity)
        }

        override fun onClick(view: View) {
            if (view.id == R.id.three_more) {
                more()
            } else {
                mHandler!!.postDelayed(Runnable {
                    val infos = HashMap<Long, MusicEntity>()
                    val len = mList!!.size
                    val list = LongArray(len)
                    for (i in 0 until len) {
                        val info = mList!![i]
                        list[i] = info.songId
                        infos.put(list[i], info)
                    }
                    if (adapterPosition >= 0) {
                        MusicPlayer.playAll(infos, list, adapterPosition, false)
                    }
                }, 70)
            }
        }
    }
}